repository_service_tuf_worker/repository.py

```diff
@@ -977,43 +978,49 @@ def bump_online_roles(self, force: Optional[bool] = False) -> bool:
 
         return True
 
-    def metadata_rotation(
+    def _trusted_root_update(
+        self, current_root: Metadata[Root], new_root: Metadata[Root]
+    ):
+        """Verify if the new metadata is a trusted Root metadata"""
+        # Verify that new root is signed by trusted root
+        current_root.verify_delegate(Root.type, new_root)
+
+        # Verify that new root is signed by itself
+        new_root.verify_delegate(Root.type, new_root)
+
+        # Verify the new root version
+        if new_root.signed.version != current_root.signed.version + 1:
+            raise BadVersionNumberError(
+                f"Expected root version {current_root.signed.version + 1}"
+                f" instead got version {new_root.signed.version}"
+            )
+
+        # Verify the Type
+        if new_root.signed.type != Root.type:
```

Comment:
Any reason to fail so late? You could check this right after Metadata.from_dict, before fetching current root from storage or verifying any signatures.

Reply:
I think the reason was to put all
checks together. But we can check it once we receive the payload and load the new root.

I don't have a preference, to be honest.

Reply:
I didn't move it out of the function but added it as a first check.